src/compass/s1_geocode_slc.py

```diff
@@ -167,68 +152,120 @@ def run(cfg: GeoRunConfig):
+                geo_ds = init_geocoded_dataset(grid_group, pol, geo_grid,
+                                               'complex64',
+                                               f'{pol} geocoded CSLC image',
+                                               output_cfg=cfg.output_params)
+                geo_datasets.append(geo_ds)
+
+                # Init geocoded output blocks/arrays lists to NaN
+                geo_data_blks.append(
+                    np.full(out_shape, np.nan + 1j * np.nan).astype(np.complex64))
+
+            dt_prep = get_time_delta_str(t_prep)
+
+            # Iterate over geogrid blocks that have radar data
+            t_geocoding = time.perf_counter()
+
+            # Declare names, types, and descriptions of carrier and flatten
+            # outputs
+            phase_names = ['azimuth_carrier_phase', 'flattening_phase']
+            phase_descrs = [f'{pol} geocoded CSLC image {desc}'
+                            for desc in phase_names]
+
+            # Prepare arrays and datasets for carrier phase and flattening
+            # phase
+            ((carrier_phase_data_blk, carrier_phase_ds),
+             (flatten_phase_data_blk, flatten_phase_ds)) = \
+            [(np.full(out_shape, np.nan).astype(np.float64),
+                  init_geocoded_dataset(grid_group, ds_name, geo_grid,
+                                        np.float64, ds_desc,
+                                        output_cfg=cfg.output_params))
+                 for ds_name, ds_desc in zip(phase_names, phase_descrs)]
+
+            # Geocode
+            isce3.geocode.geocode_slc(geo_data_blocks=geo_data_blks,
+                                      rdr_data_blocks=rdr_data_blks,
+                                      dem_raster=dem_raster,
+                                      radargrid=radar_grid,
+                                      geogrid=geo_grid, orbit=orbit,
+                                      native_doppler=native_doppler,
+                                      image_grid_doppler=image_grid_doppler,
+                                      ellipsoid=ellipsoid,
+                                      threshold_geo2rdr=threshold,
+                                      num_iter_geo2rdr=iters,
+                                      sliced_radargrid=sliced_radar_grid,
+                                      first_azimuth_line=0,
+                                      first_range_sample=0,
+                                      flatten=flatten, reramp=True,
+                                      az_carrier=az_carrier_poly2d,
+                                      rg_carrier=isce3.core.Poly2d(np.array([0])),
+                                      az_time_correction=az_lut,
+                                      srange_correction=rg_lut,
+                                      carrier_phase_block=carrier_phase_data_blk,
+                                      flatten_phase_block=flatten_phase_data_blk)
+
```

Comment:
```suggestion
            flatten_phase_data_blk = (flatten_phase_data_blk + np.pi) % (2*np.pi) - np.pi
            carrier_phase_data_blk = (carrier_phase_data_blk + np.pi) % (2*np.pi) - np.pi
```

@vbrancat The code that I'm suggesting here is to wrap the flattening phases and carrier phases before truncating the mantissa. Once this is committed to this PR, then the geolocation analysis result will get better than truncating the unwrapped phases. I think you do not need to wait until we have the final form of the data optimization in place for your AGE analysis for array mode.

@LiangJYu Can you take a look and apply the changes if it makes sense, so that Virginia can start using the code? Thanks.